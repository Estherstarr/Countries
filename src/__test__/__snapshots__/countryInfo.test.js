import React from 'react';
import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import store from '../../app/store';
import CountryInfo from '../../pages/CountryInfo';

describe('display countryInfo components', () => {
  test('test if countryInfo component renders in the dom', () => {
    render(
      <Provider store={store}>
        <Router>
          <CountryInfo />
        </Router>
      </Provider>,
    );
    expect(screen.queryByText(/Where is the world?/)).toBeNull();
  });
  test('snapshot renders properly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Router>
            <CountryInfo />
          </Router>
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
