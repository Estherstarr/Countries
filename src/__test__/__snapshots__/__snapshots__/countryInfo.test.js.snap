// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`display countryInfo components snapshot renders properly 1`] = `
<div
  className=" p-10 md:p-20 text-white md:h-screen"
>
  <a
    href="/"
    onClick={[Function]}
  >
    <button
      className="w-28 shadow-lg bg-[#2d4474] rounded p-1 mb-10"
      type="button"
    >
      Back
    </button>
  </a>
  <div
    className="flex flex-col gap-y-16 md:flex-row md:gap-x-20 "
  >
    <img
      alt="country-flag"
    />
    <div>
      <h2
        className="font-bold mb-6"
      />
      <div
        className="flex flex-col gap-y-10 md:flex-row md:gap-x-20"
      >
        <div>
          <p>
            <span
              className="font-bold"
            >
              Native Name
            </span>
            :Belge
          </p>
          <p>
            <span
              className="font-bold"
            >
               Population:
            </span>
            <span
              className="pl-2"
            />
          </p>
          <p>
            <span
              className="font-bold"
            >
              Region:
            </span>
            <span
              className="pl-2"
            />
          </p>
          <p>
            <span
              className="font-bold"
            >
              SubRegion:
            </span>
            <span
              className="pl-2"
            />
          </p>
          <p>
            <span
              className="font-bold"
            >
              Capital:
            </span>
            <span
              className="pl-2"
            />
          </p>
        </div>
      </div>
      <div
        className="mt-10 flex-col gap-y-3 md:flex-row md:gap-x-3"
      >
        <h4>
          Border Countries:
        </h4>
        <div
          className="mt-2"
        />
      </div>
    </div>
  </div>
</div>
`;
